File: saksoversikt/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/sak/service/FodselnummerAktorService.java
package no.nav.sbl.dialogarena.modiabrukerdialog.sak.service;

import no.nav.modig.core.exception.SystemException;
import no.nav.sbl.dialogarena.modiabrukerdialog.sak.providerdomain.FeilendeBaksystemException;
import no.nav.tjeneste.virksomhet.aktoer.v1.AktoerPortType;
import no.nav.tjeneste.virksomhet.aktoer.v1.meldinger.HentAktoerIdForIdentRequest;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import org.springframework.beans.factory.annotation.Autowired;

import static no.nav.sbl.dialogarena.modiabrukerdialog.sak.providerdomain.Baksystem.AKTOER;

public class FodselnummerAktorService {

    private static final Logger logger = LoggerFactory.getLogger(FodselnummerAktorService.class);

    @Autowired
    private AktoerPortType aktoerPortType;

    public String hentAktorIdForFnr(String fodselsnummer) {
        try {
            HentAktoerIdForIdentRequest request = new HentAktoerIdForIdentRequest();
            request.setIdent(fodselsnummer);
            return aktoerPortType.hentAktoerIdForIdent(request).getAktoerId();
        } catch (RuntimeException e) {
            logger.error("Det skjedde en uventet feil mot Aktoerservice", e);
            throw new FeilendeBaksystemException(AKTOER);
        } catch (Exception e) {
            throw new SystemException("Feil ved henting av aktorId for fnr: " + fodselsnummer, e);
        }
    }
}

File: web/src/main/java/no/nav/modiapersonoversikt/service/FodselnummerAktorService.kt
package no.nav.modiapersonoversikt.service

import no.nav.modiapersonoversikt.legacy.api.service.pdl.PdlOppslagService
import org.springframework.beans.factory.annotation.Autowired

interface FodselnummerAktorService {
    fun hentAktorIdForFnr(fnr: String): String?
    fun hentFnrForAktorId(aktorId: String): String?
}

open class FodselnummerAktorServiceImpl @Autowired constructor(
    val pdlOppslagService: PdlOppslagService
) : FodselnummerAktorService {
    override fun hentAktorIdForFnr(fnr: String): String {
        return requireNotNull(pdlOppslagService.hentAktorId(fnr)) {
            "Feil ved henting av aktorId for $fnr"
        }
    }

    override fun hentFnrForAktorId(aktorId: String): String {
        return requireNotNull(pdlOppslagService.hentFnr(aktorId)) {
            "Feil ved henting av fnr for $aktorId"
        }
    }
}
